src/main/java/io/druid/segment/loading/DataSegmentPuller.java

```diff
@@ -10,10 +13,49 @@
 {
   /**
    * Pull down segment files for the given DataSegment and put them in the given directory.
+   * Deprecated in favor of URI pulling when possible
    *
    * @param segment The segment to pull down files for
-   * @param dir The directory to store the files in
+   * @param dir     The directory to store the files in
+   *
    * @throws SegmentLoadingException if there are any errors
    */
+  @Deprecated
   public void getSegmentFiles(DataSegment segment, File dir) throws SegmentLoadingException;
+
+  /**
+   * Pull down segment files for the given DataSegment and put them in the given directory.
+   *
+   * @param segmentUri The URI for the segment
+   * @param dir        The directory to store the files in
+   *
+   * @throws SegmentLoadingException if there are any errors
+   */
+  public void getSegmentFiles(URI segmentUri, File dir) throws SegmentLoadingException;
```

Comment:
>Also, can you explain why we can't just pass around the segment object (versus using the loadSpec)

Part of my problem is that Segment objects have a lot of extra baggage  that may not be globally appropriate for pulling general data down. For example, a segment assumes intervals, metrics, and dimensions.

It seems to me that a Segment is a special case of FlatData (a term I'm using for the first time now) rather than everything being a case of Segment.